modules/social_features/social_swiftmail/social_swiftmail.module

```diff
@@ -69,115 +90,86 @@ function social_swiftmail_preprocess_swiftmailer(array &$variables) {
+    $variables['site_sub_slogan'] = Link::fromTextAndUrl(t('#MadeToShare'), Url::fromUri('https://twitter.com/search?q=%23MadeToShare%20%23opensocial'));
   }
 
+  /** @var \Drupal\Core\Render\RendererInterface $renderer */
+  $renderer = \Drupal::service('renderer');
+
   // Check if a custom e-mail header is set and apply the configuration
   // to the render array.
-  if (($header_config = $social_swiftmail_config->get('template_header'))
+  if (($header_config = $email_config->get('template_header'))
     && (!empty($header_config['value']) && !empty($header_config['format']))) {
     $header_markup = [
       '#type' => 'processed_text',
       '#text' => $header_config['value'],
       '#format' => $header_config['format'],
     ];
-    $variables['header'] = \Drupal::service('renderer')->renderRoot($header_markup);
+    $variables['header'] = $renderer->renderRoot($header_markup);
   }
 
   // Check if a custom e-mail footer is set and apply the configuration
   // to the render array.
-  if (($footer_config = $social_swiftmail_config->get('template_footer'))
+  if (($footer_config = $email_config->get('template_footer'))
     && (!empty($footer_config['value']) && !empty($footer_config['format']))) {
     $footer_markup = [
       '#type' => 'processed_text',
       '#text' => $footer_config['value'],
       '#format' => $footer_config['format'],
     ];
-    $variables['footer'] = \Drupal::service('renderer')->renderRoot($footer_markup);
+    $variables['footer'] = $renderer->renderRoot($footer_markup);
   }
 
-  $user = FALSE;
-  $message = &$variables['message'];
+  /** @var \Drupal\symfony_mailer\EmailInterface $email */
+  $email = $variables['email'];
+  $context = $email->getParam('context');
 
-  if (!empty($message['to'])) {
-    $user = user_load_by_mail($message['to']);
+  if ($to = $email->getHeaders()->getHeaderBody('X-Rerouted-Original-to')) {
+    $mail = $to;
   }
-
-  // Try to add a heading message.
-  if ($user || !empty($message['params']['context']['display_name'])) {
-    $options = [
-      'langcode' => $message['langcode'],
-    ];
-
-    if ($user) {
-      $display_name = $user->getDisplayName();
-    }
-    else {
-      $display_name = $message['params']['context']['display_name'];
-    }
-
-    $disabled_greeting_keys = $social_swiftmail_config->get("disabled_user_greeting_keys");
-    $disabled_greeting_keys = explode("\r\n", $disabled_greeting_keys ?? "");
-    $replace = [
-      '@display_name' => $display_name,
-    ];
-
-    if (isset($message['key'])) {
-      if (!in_array($message['key'], $disabled_greeting_keys)) {
-        $variables['heading'] = t('Hi <strong>@display_name</strong>', $replace, $options);
-      }
-      return;
-    }
-
-    $variables['heading'] = t('Hi <strong>@display_name</strong>', $replace, $options);
+  else {
+    $to = $email->getTo();
+    $mail = isset($to[0]) ? _addresses_to_string($to)[0] : FALSE;
```

Comment:
Seems like this could do with some explanation what the if / else is trying to achieve